libraries/suid_check.rb

```diff
@@ -6,21 +6,44 @@ class SUIDCheck < Inspec.resource(1)
   name 'suid_check'
   desc 'Use the suid_check resource to verify the current SUID/SGID against a blacklist'
   example "
-     describe suid_check(blacklist) do
+     describe suid_check(blacklist, exclude_fstypes: %w[nfs nfs4], exclude_paths: %w[/proc]) do
        its('diff') { should be_empty }
      end
    "
 
-  def initialize(blacklist = nil)
-    blacklist = default if blacklist.nil?
+  def initialize(blacklist = nil, exclude_fstypes: nil, exclude_paths: nil)
     @blacklist = blacklist
+    @exclude_fstypes = exclude_fstypes || default_exclude_fstypes
+    @exclude_paths = exclude_paths || default_exclude_paths
+  end
+
+  def default_exclude_fstypes
```

Comment:
By adding those, we change the default behaviour, which results in a breaking change. @arlimus @atomic111 we have two options:
- release a new major version
- keep the old behaviour but allow the override
What do you think?

Reply:
My $0.02 is that it should be the default behavior and have a major version bump because the current default of walking network filesystems is prohibitively expensive and redundant when you have a lot of nodes mounting the same share. If you're using the audit cookbook to run this profile in that scenario, some nodes could take longer than the Chef run interval to even complete. If you're using push jobs on large groups, the job will take as long as the longest node, so using the current version of this profile has increased deploy times as well. IMO, if you want to check network shares for SUID, that should be done by a single ad-hoc process somewhere.